scripts/ubuntu14-mitaka-to-newton.sh

```diff
@@ -0,0 +1,219 @@
+  # so we can get the appropriate basename for the upgrade_marker
+  for part in $run_item; do
+    if [[ "$part" == *.yml ]];then
+      file_part="$part"
+      break
+    fi
+  done
+
+  if [ ! -d  "/etc/openstack_deploy/upgrade-${TARGET_SERIES}" ]; then
+      mkdir -p "/etc/openstack_deploy/upgrade-${TARGET_SERIES}"
+  fi
+
+  upgrade_marker_file=$(basename ${file_part} .yml)
+  upgrade_marker="/etc/openstack_deploy/upgrade-${TARGET_SERIES}/$upgrade_marker_file.complete"
+
+  if [ ! -f "$upgrade_marker" ];then
+    # note(sigmavirus24): use eval so that we properly turn strings like
+    # "/tmp/fix_container_interfaces.yml || true"
+    # into a command, otherwise we'll get an error that there's no playbook
+    # named ||
+    eval "openstack-ansible $2"
+    playbook_status="$?"
+    echo "ran $run_item"
+
+    if [ "$playbook_status" == "0" ];then
+      RUN_TASKS=("${RUN_TASKS[@]/$run_item}")
+      touch "$upgrade_marker"
+      echo "$run_item has been marked as success"
+    else
+      echo "******************** failure ********************"
+      echo "The upgrade script has encountered a failure."
+      echo "Failed on task \"$run_item\""
+      echo "Re-run the run-upgrade.sh script, or"
+      echo "execute the remaining tasks manually:"
+      # NOTE:
+      # List the remaining, incompleted tasks from the tasks array.
+      # Using seq to genertate a sequence which starts from the spot
+      # where previous exception or failures happened.
+      # run the tasks in order
+      for item in $(seq $1 $((${#RUN_TASKS[@]} - 1))); do
+        if [ -n "${RUN_TASKS[$item]}" ]; then
+          echo "openstack-ansible ${RUN_TASKS[$item]}"
+        fi
+      done
+      echo "******************** failure ********************"
+      exit 99
+    fi
+  else
+    RUN_TASKS=("${RUN_TASKS[@]/$run_item.*}")
+  fi
+  set -e
+}
+
+function pre_cleanup {
+  # remove all ansible_ssh_host entries
+  sed -i '/ansible_host/d' /etc/openstack_deploy/user*.yml
+  # upgrade looks for user_variables so drop one in place for upgrade
+  if [[ ! -f /etc/openstack_deploy/user_variables.yml ]]; then
+     echo "---" > /etc/openstack_deploy/user_variables.yml
+     echo "default_bind_mount_logs: False" >> /etc/openstack_deploy/user_variables.yml
+  elif [[ -f /etc/openstack_deploy/user_variables.yml ]]; then
+     echo "default_bind_mount_logs: False" >> /etc/openstack_deploy/user_variables.yml
+  fi
+}
+
+function check_for_current {
+    if [[ ! -d "/etc/openstack_deploy" ]]; then
+      echo "--------------ERROR--------------"
+      echo "/etc/openstack_deploy directory not found."
+      echo "It appears you do not have a current environment installed."
+      exit 2
+    fi
+}
+
+function pre_flight {
+    ## Library Check -------------------------------------------------------------
+
+    info_block "Checking for required libraries." 2> /dev/null ||
+        source ${SCRIPTS_PATH}/scripts-library.sh
```

Comment:
scripts/ubuntu14-mitaka-to-newton.sh: line 135: /opt/rpc-upgrades/scripts/scripts-library.sh: No such file or directory